File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let curr=head;
    let count=0;
    while(curr!=null){
        count++;
        curr=curr.next;
    }
    let pos=count-n;
    curr=head;
    if(pos==0)return head.next;
    for(let i=0;i<pos-1;i++){
        curr=curr.next;
    }
    if(curr.next!=null){
    curr.next=curr.next.next;}
    return head;
    
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
       ListNode* curr=head;
       int count=0;
       while(curr!=NULL){
        count++;
        curr=curr->next;
       }
       if(n==count){
        return head->next;
       }
       int k=count-n;
       ListNode* temp=head;
       for(int i=1;i<k;i++){
            temp=temp->next;
       }
       temp->next=temp->next->next;
       return head;

    }
};